File: remove_bg.py
import sys
import os

input_img = r"C:\Users\안기정\.gemini\antigravity\brain\821c9c05-11cf-4c1f-baf4-23f4650d39e1\media__1771833273276.jpg"
output_img = r"c:\dev\department-b\public\logo.png"

try:
    import rembg
    from PIL import Image
    with open(input_img, "rb") as i:
        with open(output_img, "wb") as o:
            input_data = i.read()
            # Post-processing to avoid hard edges or using basic rembg
            output_data = rembg.remove(input_data)
            o.write(output_data)
    print("Background removed using rembg")
except ImportError:
    print("rembg not found, falling back to PIL")
    try:
        from PIL import Image
        def remove_white(input_path, output_path, tolerance=210):
            img = Image.open(input_path).convert("RGBA")
            datas = img.getdata()
            
            newData = []
            for item in datas:
                brightness = (item[0]+item[1]+item[2])/3
                if brightness > tolerance:
                    if brightness > 245:
                        newData.append((255,255,255,0))
                    else:
                        alpha = int(255 - (brightness - tolerance) * (255/(245-tolerance)))
                        if alpha < 0: alpha = 0
                        newData.append((item[0], item[1], item[2], alpha))
                else:
                    newData.append(item)
                    
            img.putdata(newData)
            img.save(output_path, "PNG")
        
        remove_white(input_img, output_img)
        print("Background removed using PIL")
    except Exception as e:
        print(f"Failed to use PIL: {e}")

File: remove_bg.js
const Jimp = require('jimp');

async function removeWhite() {
    try {
        const imagePath = String.raw`C:\Users\안기정\.gemini\antigravity\brain\821c9c05-11cf-4c1f-baf4-23f4650d39e1\media__1771833273276.jpg`;
        const outputPath = String.raw`c:\dev\department-b\public\logo.png`;

        const image = await Jimp.read(imagePath);

        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
            const r = this.bitmap.data[idx + 0];
            const g = this.bitmap.data[idx + 1];
            const b = this.bitmap.data[idx + 2];

            const brightness = (r + g + b) / 3;
            const isGray = Math.abs(r - g) < 20 && Math.abs(g - b) < 20 && Math.abs(r - b) < 20;

            if (brightness > 230 && isGray) {
                if (brightness > 245) {
                    this.bitmap.data[idx + 3] = 0;
                } else {
                    let alpha = 255 - ((brightness - 230) * (255 / 15));
                    if (alpha < 0) alpha = 0;
                    this.bitmap.data[idx + 3] = Math.floor(alpha);
                }
            }
        });

        await image.writeAsync(outputPath);
        console.log("Successfully processed and saved logo.png");
    } catch (error) {
        console.error("Error processing image:", error);
    }
}

removeWhite();
